sql/src/main/java/com/vmware/ddlog/translator/JoinInfo.java

```diff
@@ -38,32 +38,68 @@
 
 /**
  * This visitor inspects the predicate of a join to detect whether it is an equijoin.
- * If it is, it maintains a list of pairs of columns checked for equivalence.
+ * If it is, it maintains information about the columns that participate in the join.
+ * The values below are for the example in TranslationVisitor.visitJoin.
  */
-public class JoinColumns extends AstVisitor<Void, Void> {
-    // Maps a column in the right table to the equal column in the left table.
+public class JoinInfo extends AstVisitor<Void, Void> {
+    /**
+     * Maps a column in the right table to the equal column in the left table.
+     * Contents: c1 -> c1, c2 -> c3.
+     * If this is null the join is not an equijoin.
+     */
     @Nullable
     protected HashMap<String, String> rightToLeftColumn;
-    protected final HashSet<String> leftColumns;
-    // Type of each column
-    protected HashMap<String, DDlogType> leftColumnType;
-    // For each column in the left table keep here the variable name used in the code to match it
-    protected HashMap<String, String> leftColumnVariable;
-    protected EnvHandle leftEnvironment;
-    protected EnvHandle rightEnvironment;
+    /**
+     * Maps a column name in the left table to the equal column in the right table.
+     * c1 -> c1, c3 -> c2.
+     */
+    protected final HashMap<String, String> leftToRightColumn;
```

Comment:
Odd to not use the interface types in the declarations (Map, Set) instead of the implementation (HashMap, HashSet)